main.py

```diff
@@ -0,0 +1,56 @@
+import requests
+from dotenv import dotenv_values
+import pathlib
+
+def get_metadata(token, path_dir):
+        
+    my_headers = {'Authorization' : f'Bearer {token}'}
+    response = requests.get('https://api.github.com/user/repos', headers=my_headers)
```

Comment:
Take a look if you can get the pagination from the request so you can loop over all pages to get all records from requests. Once your request from the current page get's no records back, stop the loop.